inference-engine/src/transformations/src/transformations/convert_quantize_dequantize.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/convert_quantize_dequantize.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+
+ngraph::pass::ConvertQuantizeDequantize::ConvertQuantizeDequantize() {
+    auto data_pattern = ngraph::pattern::any_input();
+    auto input_low_pattern = ngraph::pattern::any_input();
+    auto input_high_pattern = ngraph::pattern::any_input();
+    auto output_low_pattern = ngraph::pattern::wrap_type<opset1::Constant>();
+    auto output_high_pattern = ngraph::pattern::wrap_type<opset1::Constant>();
+    auto fq_pattern = ngraph::pattern::wrap_type<opset1::FakeQuantize>({data_pattern, input_low_pattern,
+                                                                       input_high_pattern, output_low_pattern,
+                                                                       output_high_pattern});
+    auto convert1_pattern = ngraph::pattern::wrap_type<opset1::Convert>({fq_pattern});
+    auto convert2_pattern = ngraph::pattern::wrap_type<opset1::Convert>({convert1_pattern});
+    auto zero_point_pattern = ngraph::pattern::any_input();
+    auto sub_pattern = ngraph::pattern::wrap_type<opset1::Subtract>({convert2_pattern, zero_point_pattern}, pattern::consumers_count(1));
+    auto scale_pattern = ngraph::pattern::any_input();
+    auto mul_pattern = ngraph::pattern::wrap_type<opset1::Multiply>({sub_pattern, scale_pattern});
+
+    ngraph::graph_rewrite_callback callback = [=](pattern::Matcher& m) {
+        auto mul = std::dynamic_pointer_cast<ngraph::opset1::Multiply> (m.get_match_root());
+        if (!mul)
+            return false;
```

Comment:
I mean is there any chance to have nullptr here if the pattern matches to Multiply only? I just don't want to see the same action in each callback if we guarantee that the pattern is matched. At least use assert instead of if-statement.

Reply:
yes we can rely on matcher and we can avoid this check if we do not cast node  ptr.